frontend/packages/shared/src/navigation/main-header/ThreeDotsMenu.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState, useEffect, useRef } from 'react';
+import Classes from './ThreeDotsMenu.module.css';
+import { Settings } from '@navikt/ds-icons';
+import { Link, useParams } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { FaRegClone } from 'react-icons/fa';
+import { CloneModal } from 'app-development/layout/version-control/CloneModal';
+import { repositoryPath } from 'app-shared/api-paths';
+
+export function ThreeDotsMenu() {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const menuRef = useRef(null);
+  const [cloneModalAnchor, setCloneModalAnchor] = useState(null);
+  const { org, app } = useParams();
+  const { t } = useTranslation();
+  const closeCloneModal = () => setCloneModalAnchor(null);
+  const openCloneModal = (event: React.MouseEvent) => setCloneModalAnchor(event.currentTarget);
+
+  function handleButtonClick() {
+    setIsMenuOpen(!isMenuOpen);
+  }
+
+  function handleCloneOutsideClick(event: MouseEvent) {
+    if (menuRef.current && !menuRef.current.contains(event.target)) {
+      setIsMenuOpen(false);
+    } else cloneModalAnchor && !cloneModalAnchor.contains(event.target);
+    {
+      closeCloneModal();
+    }
+  }
+
+  useEffect(() => {
+    {
+      if (setIsMenuOpen) {
+        document.addEventListener('mousedown', handleCloneOutsideClick);
```

Comment:
Dette vil legge til en ny event listener hver gang komponenten laster på nytt. For å håndtere dette, bør du legge til `return () => document.removeEventListener('mousedown', handleCloneOutsideClick)`. Funksjonen som returneres av en `useEffect` kjøres på slutten av komponentens livssyklus. Du kan lese om dette her: https://react.dev/reference/react/useEffect

Uansett bør du vurdere om du kan bruke `Popover` fra designsystemet til å løse dette; https://digdir.github.io/designsystem/?path=/docs/kjernekomponenter-popover--popover

Reply:
fikset